apps/omg_child_chain/test/omg_child_chain/event_fetcher_test.exs

```diff
@@ -0,0 +1,497 @@
+      {to_block, get_signature_from_event(events, :deposit_created), []},
+      {to_block, get_signature_from_event(events, :in_flight_exit_output_piggybacked), []},
+      {to_block, get_signature_from_event(events, :exit_started), [exit_started_log]},
+      {to_block, get_signature_from_event(events, :in_flight_exit_input_piggybacked),
+       [in_flight_exit_input_piggybacked_log]},
+      {to_block, get_signature_from_event(events, :in_flight_exit_started), []}
+    ]
+
+    # we're just making sure that handle continue gets called after handle_call
+    :ok = Process.sleep(100)
+    assert :ets.tab2list(table) -- what_should_be_left_in_db == []
+  end
+
+  # We also assert if blocks that did NOT have any events get commited to ETS as empty.
+  # This is important because we do not want to re-scan blocks for which we know contain nothing.
+  test "if we get response for range and all events are commited to ETS", %{
+    event_fetcher_name: event_fetcher_name,
+    table: table,
+    test: test_name
+  } do
+    defmodule test_name do
+      alias OMG.ChildChain.EventFetcherTest
+
+      def get_ethereum_events(from_block, to_block, _signatures, _contracts) do
+        {:ok,
+         [
+           EventFetcherTest.deposit_created_log(from_block),
+           EventFetcherTest.deposit_created_log(from_block + 1),
+           EventFetcherTest.exit_started_log(to_block),
+           EventFetcherTest.in_flight_exit_output_piggybacked_log(from_block),
+           EventFetcherTest.in_flight_exit_input_piggybacked_log(to_block)
+         ]}
+      end
+
+      def get_call_data(_tx_hash) do
+        {:ok, EventFetcherTest.start_standard_exit_log()}
+      end
+    end
+
+    :sys.replace_state(event_fetcher_name, fn state -> Map.put(state, :rpc, test_name) end)
+    events = event_fetcher_name |> :sys.get_state() |> Map.get(:events)
+
+    from_block = 1
+    to_block = 3
+    deposit_created = from_block |> deposit_created_log() |> Abi.decode_log()
+    deposit_created_2 = from_block |> Kernel.+(1) |> deposit_created_log() |> Abi.decode_log()
+
+    assert EventFetcher.deposit_created(event_fetcher_name, from_block, to_block) ==
+             {:ok, [deposit_created, deposit_created_2]}
+
+    exit_started_log =
+      to_block
+      |> exit_started_log()
+      |> Abi.decode_log()
+      |> Map.put(:call_data, start_standard_exit_log() |> from_hex |> Abi.decode_function())
+
+    assert EventFetcher.exit_started(event_fetcher_name, from_block, to_block) == {:ok, [exit_started_log]}
+
+    in_flight_exit_output_piggybacked_log = from_block |> in_flight_exit_output_piggybacked_log() |> Abi.decode_log()
+    in_flight_exit_input_piggybacked_log = to_block |> in_flight_exit_input_piggybacked_log() |> Abi.decode_log()
+
+    assert EventFetcher.in_flight_exit_piggybacked(event_fetcher_name, from_block, to_block) ==
+             {:ok, [in_flight_exit_input_piggybacked_log, in_flight_exit_output_piggybacked_log]}
+
+    assert :ets.tab2list(table) --
```

Comment:
can these `--` assertions be substituted by regular `==` assertions? `[]` would satisfy this assertion for example

Reply:
I'm not sure I exactly understand this comment but let me explain what the assertion does:
I'm asserting that the contents of the ETS table reduced by the expected content is an empty list. so I'm just verifying what I've written to the DB.

Reply:
I think you're rather asserting that the ETS table doesn't contain entries absent in the big list given.
If `:ets.tab2list(table)` returns `[]` this assertion passes, is this intended? (if it is then I'm missing something).

```
iex(2)> assert [] -- [1] == []
true
```

Apart from that, I find an `assert :ets.tab2list(table) == some_expected_value` much more readable, if your assertions can be rewritten to this.

Reply:
oh I think you've nailed it here. fixed it!